client/src/plugins/user-journey-statistics/event-handlers/ModelingEventHandler.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.
+ *
+ * Camunda licenses this file to you under the MIT; you may not use this file
+ * except in compliance with the MIT License.
+ */
+
+import { isModelElement } from 'diagram-js/lib/model';
+import { getBusinessObject } from 'bpmn-js/lib/util/ModelUtil';
+
+export default class ModelingEventHandler {
+  constructor(props) {
+
+    const {
+      subscribe,
+      track
+    } = props;
+
+    this.track = track;
+    this.bpmnJSTracking = null;
+
+    this.subscribeToModelingEvents(subscribe);
+
+    subscribe('telemetry.enabled', () => {
+      if (this.bpmnJSTracking) {
+        this.bpmnJSTracking.enable();
+      }
+    });
+
+    subscribe('telemetry.disabled', () => {
+      if (this.bpmnJSTracking) {
+        this.bpmnJSTracking.disable();
+      }
+    });
+  }
+
+  subscribeToModelingEvents = (subscribe) => {
+
+    subscribe('bpmn.modeler.created', async (event) => {
+      const { modeler } = event;
+
+      this.bpmnJSTracking = modeler.get('bpmnJSTracking');
+
+      this.bpmnJSTracking.enable();
+
+      this.bpmnJSTracking.on('tracking.event', (trakingEvent) => {
+        const { name, data } = trakingEvent;
+
+        // format event name to be consistent
+        const eventName = name.replace('.', ':');
+
+        // replace moddle elements with serializable data
+        const serializableData = replaceModleElements(data);
+
+        this.track(eventName, serializableData);
+      });
+    });
+  };
+
+}
+
+
+// helpers //////////
+
+function replaceModleElement(object) {
```

Comment:
```suggestion
function replaceModdleElement(object) {
```

Are these `moddle` (businessObject) or diagram elements?

Reply:
Right now it's the whole diagram element; we send `element: {id, type, template?}`. We may decide to extend the data we send in the future (or now already if we see fit), for example considering event definitions and such. I decided to keep it short and expand as needed. Also why @christian-konrad is cc'ed here.

Reply:
Let's rename this method to `filterDiagramElement` then or something.